common/persistence/client/factory.go

```diff
@@ -84,18 +85,20 @@ func NewFactory(
 	cfg *config.Persistence,
 	ratelimiter quotas.RequestRateLimiter,
 	serializer serialization.Serializer,
+	historyBranchUtil p.HistoryBranchUtil,
```

Comment:
If we keep this as an object instead of a set of utility functions, we should remove the parameter and construct it in line here. For now, there is only one implementation of HistoryBranchUtil, which has no state or constructor parameters.

Reply:
This is an extension point for injecting different logic.